renderdoc/driver/d3d12/d3d12_shader_feedback.cpp

```diff
@@ -633,41 +810,94 @@ static bool AnnotateDXILShader(const DXBC::DXBCContainer *dxbc, uint32_t space,
         DXIL::Value(editor.GetOrAddConstant(f, DXIL::Constant(i32, magicFeedbackValue))),
     };
 
-    editor.AddInstruction(f, 1, inst);
+    editor.AddInstruction(f, startInst++, inst);
   }
 
-  for(size_t i = 2; i < f->instructions.size(); i++)
+  for(size_t i = startInst; i < f->instructions.size(); i++)
   {
     const DXIL::Instruction &inst = f->instructions[i];
     // we want to annotate any calls to createHandle
-    if(inst.op == DXIL::Operation::Call && inst.funcCall->name == createHandle->name)
+    if(inst.op == DXIL::Operation::Call &&
+       ((createHandle && inst.funcCall->name == createHandle->name) ||
+        (createHandleFromBinding && inst.funcCall->name == createHandleFromBinding->name)))
     {
-      if(inst.args.size() != 5)
+      DXIL::Value idxArg;
+      rdcpair<uint32_t, int32_t> slotInfo = {0, 0};
+      if((createHandle && inst.funcCall->name == createHandle->name))
       {
-        RDCERR("Unexpected number of arguments to createHandle");
-        continue;
-      }
+        RDCASSERTEQUAL(isShaderModel6_6OrAbove, false);
+        if(inst.args.size() != 5)
+        {
+          RDCERR("Unexpected number of arguments to createHandle");
+          continue;
+        }
 
-      DXIL::Value kindArg = inst.args[1];
-      DXIL::Value idArg = inst.args[2];
-      DXIL::Value idxArg = inst.args[3];
+        DXIL::Value kindArg = inst.args[1];
+        DXIL::Value idArg = inst.args[2];
+        idxArg = inst.args[3];
 
-      if(kindArg.type != DXIL::ValueType::Constant || idArg.type != DXIL::ValueType::Constant)
-      {
-        RDCERR("Unexpected non-constant argument to createHandle");
-        continue;
+        if(kindArg.type != DXIL::ValueType::Constant || idArg.type != DXIL::ValueType::Constant)
+        {
+          RDCERR("Unexpected non-constant argument to createHandle");
+          continue;
+        }
+        DXIL::HandleKind kind = (DXIL::HandleKind)kindArg.constant->val.u32v[0];
+        uint32_t id = idArg.constant->val.u32v[0];
+        if(kind == DXIL::HandleKind::SRV && id < srvBaseSlots.size())
+          slotInfo = srvBaseSlots[id];
+        else if(kind == DXIL::HandleKind::UAV && id < uavBaseSlots.size())
+          slotInfo = uavBaseSlots[id];
       }
+      else
+      {
+        RDCASSERTEQUAL(isShaderModel6_6OrAbove, true);
+        if(inst.args.size() != 4)
+        {
+          RDCERR("Unexpected number of arguments to createHandleFromBinding");
+          continue;
+        }
+        DXIL::Value resBindArg = inst.args[1];
+        idxArg = inst.args[2];
+        if(resBindArg.type != DXIL::ValueType::Constant)
+        {
+          RDCERR("Unexpected non-constant argument to createHandleFromBinding");
+          continue;
+        }
+        // resbind can sometime be empty
```

Comment:
Looking at it in more details, it seems this case is handled incorrectly, resbind is empty when the variable is nullconst, so it should be handled as "all args to 0" instead of being skipped

Reply:
Ahh, that makes sense as the dxil reader doesn't currently 'expand' null constants of aggregate type like that. In future that might be something to consider, but dxil is absolutely awful to work with so better to handle it deliberately here exactly as you say.